src/GuardClauses/GuardClauseExtensions.cs

```diff
@@ -440,9 +438,9 @@ public static void NegativeOrZero([NotNull] this IGuardClause guardClause, doubl
         /// <param name="input"></param>
         /// <param name="parameterName"></param>
         /// <exception cref="ArgumentException"></exception>
-        public static void Default<T>([NotNull] this IGuardClause guardClause, [NotNull] T input, [NotNull] string parameterName)
+        public static void Default<T>([JetBrainsNotNull] this IGuardClause guardClause, [AllowNull, JetBrainsNotNull] T input, [JetBrainsNotNull] string parameterName)
```

Comment:
Since C# does not allow the syntax `T?` for open generic types, `[AllowNull] T input` must be used rather than `[NotNull T? input`.

AllowNull, DisallowNull, MaybeNull, and NotNull override the nullability of the type they are applied to.

AllowNull and DisallowNull affect the validation of the expression passed to the location they are applied to.
MaybeNull and NotNull override the nullability of the same expression after the method returns without throwing.